mysql-test/main/mysqltest-break.test

```diff
@@ -0,0 +1,89 @@
+--echo #
+--echo # Test "break" command
+--echo #
+
+# Break in a single loop
+
+let $cnt= 4;
+while($cnt > 1)
+{
+  dec $cnt;
+  break;
+  --echo Break did not stop a single loop
+}
+
+# Break stops inner loop
+
+let $outer= 4;
+while($outer > 1)
+{
+  let $inner= 4;
+  while($inner > 1)
+  {
+    if($outer == 2)
+    {
+      --echo OK
+    }
+    if($inner == 2)
+    {
+      break;
+    }
+    dec $inner;
+  }
+  dec $outer;
+  --echo $outer
+}
+
+# Break stops outer loop
+let $inner= 4;
+let $outer= 4;
+while($outer > 1)
+{
+  break;
+  while($inner > 1)
+  {
+    dec $inner;
+    --echo Outer loop`s break did not stop inner loop
+  }
+  dec $outer;
+  --echo $outer
+}
+
+# Break stops loop in if
+let $cnt= 4;
+if($cnt > 1)
+{
+  while($cnt)
+  {
+    break;
+  }
+  dec $cnt;
+}
+
+if($cnt != 3)
+{
+  --echo Inner loop stopped outer block
+}
+
+# Break in inner if
+
+let $cnt= 4;
+while($cnt > 1)
+{
+  if($cnt == 2)
+  {
+    break;
+    --echo "if" is working after break
+  }
+  dec $cnt;
+}
+
+if($cnt == 1)
+{
+  --echo Break did not work in inner while
```

Comment:
in the spirit of other test, here you'd better do just `--echo cnt=$cnt` without any `if()` and let mysqltest detect the difference in the output if break won't work.